core/src/main/java/org/mskcc/cbio/portal/dao/DaoGene.java

```diff
@@ -265,9 +278,10 @@ private static CanonicalGene getGene(long entrezGeneId) throws DaoException {
                     ("SELECT * FROM gene");
             rs = pstmt.executeQuery();
             while (rs.next()) {
+            	int geneticEntityId = rs.getInt("GENETIC_ENTITY_ID");
                 long entrezGeneId = rs.getInt("ENTREZ_GENE_ID");
                 Set<String> aliases = mapAliases.get(entrezGeneId);
-                CanonicalGene gene = new CanonicalGene(entrezGeneId,
+                CanonicalGene gene = new CanonicalGene(geneticEntityId, entrezGeneId,
```

Comment:
I expect this to be considered an internal id .. and not be reported through a web API, since it is arbitrary and subject to change between database rebuilds.